app/scripts/controllers/swaps.js

```diff
@@ -474,30 +469,43 @@ export default class SwapsController {
         : totalEthCost
 
       const tokenConversionRate = tokenConversionRates[destinationToken]
-      const ethValueOfTrade =
+
+      const decimalAdjustedDestinationAmount = calcTokenAmount(destinationAmount, destinationTokenInfo.decimals)
+
+      const tokenValueOfQuoteForSorting =
         destinationToken === ETH_SWAPS_TOKEN_ADDRESS
-          ? calcTokenAmount(destinationAmount, 18).minus(totalEthCost, 10)
+          ? decimalAdjustedDestinationAmount.minus(totalEthCost, 10)
           : new BigNumber(tokenConversionRate || 1, 10)
-            .times(
-              calcTokenAmount(
-                destinationAmount,
-                destinationTokenInfo.decimals,
-              ),
-              10,
-            )
+            .times(decimalAdjustedDestinationAmount)
             .minus(tokenConversionRate ? totalEthCost : 0, 10)
 
-      // collect values for savings calculation
-      allEthTradeValues.push(ethValueOfTrade)
-      allEthFees.push(ethFee)
+      const ethValueOfQuote =
+        destinationToken === ETH_SWAPS_TOKEN_ADDRESS
+          ? decimalAdjustedDestinationAmount.minus(totalEthCost, 10)
+          : new BigNumber(tokenConversionRate || 0, 10)
```

Comment:
Nit: It has been a little hard to follow all of these `tokenConversionRate || 0` conditions.

Maybe this would be simpler if we split this into three branches, and had all of these calculations in them, e.g.

```
if (destinationToken === ETH_SWAPS_TOKEN_ADDRESS) {
  ...
} else if (tokenConversionRate === undefined) {
  ...
} else {
  ...
}
```